wagtail/actions/copy_page.py

```diff
@@ -249,7 +253,9 @@ def _copy_page(
             page_copy.live_revision = latest_revision_as_page_revision
             page_copy.last_published_at = latest_revision_as_page_revision.created_at
             page_copy.first_published_at = latest_revision_as_page_revision.created_at
-            page_copy.save(clean=False)
+
+        page_copy.latest_revision = latest_revision_as_page_revision
+        page_copy.save(clean=False)
```

Comment:
I was wondering why this was needed, as this new revision was created using `save_revision()`, which should update the `latest_revision` foreign key in the process.

It turns out we're deliberately working with two separate objects here:
- `page_copy`, which is the current live version of the page (a.k.a. real page record)
- `latest_revision = page_copy.get_latest_revision_as_object()`, which is the latest revision as a page object. (The variable name is confusing here now that we have an FK with the same name. I'll refer to this as "the variable", while the other as "the FK")

The `save_revision()` method was called on `latest_revision` (the variable) and not `page_copy`, which seems understandable as we want to save the changes in `update_attrs` (which may contain some updates from the Copy view, e.g. the title) into a new revision, while keeping the `page_copy` intact (case in point: copying a live+draft page with `keep_live=True`). However, this does mean that the `page_copy.latest_revision` FK didn't get updated by `save_revision()`, hence the need for these new lines.

That said, **the FK, along with other fields handled in `save_revision()` (e.g. `draft_title`, `latest_revision_created_at`) still get updated on the database**, because `page_copy` and `latest_revision` (the variable) are based off the same page record. This is hinted by the comment above this code.

Unfortunately, calling `save()` here would overwrite what was updated by `save_revision()`, namely the `draft_title` and `latest_revision_created_at`, which causes a bug in the listing. You can verify this by specifying a different title in the Copy form in the CMS, and seeing that the page listing (which uses those fields) does not show the new title (the editor works fine as it loads the latest revision as the edited object). In addition, the newly copied object is not placed at the top of the listing when ordered by the update timestamp. This means that the purposes mentioned by the comment above the code is no longer working.

I believe the correct solution is to leave `page_copy.save(clean=False)` inside the `if self.keep_live`, and move the `page_copy.latest_revision` assignment to above the `if`, e.g.

```py
        page_copy.latest_revision = latest_revision_as_page_revision
        if self.keep_live:
            page_copy.live_revision = latest_revision_as_page_revision
            page_copy.last_published_at = latest_revision_as_page_revision.created_at
            page_copy.first_published_at = latest_revision_as_page_revision.created_at
            page_copy.save(clean=False)
```

We shouldn't have to worry about updating the `latest_revision` FK if `keep_live` is `False`, as I believe the FK has been updated on the database by `save_revision()`.

It's probably worth adding assertions to test the regression, e.g.

```py
about_us_updated = SimplePage.objects.get(url_path="/home/about-us/")
self.assertEqual(about_us_updated.draft_title, "New about us")
```

That said, with the introduction of `latest_revision`, ideally we should remove the denormalised `draft_title` and `latest_revision_created_at`, and replace them with `latest_revision.object_str` (or from the revision's content) and `latest_revision.created_at`, which avoids the issue by making `latest_revision` be the single source of truth.

Reply:
> You can verify this by specifying a different title in the Copy form in the CMS, and seeing that the page listing (which uses those fields) does not show the new title

Thanks for this comment @laymonage, I was testing this PR and also observed the same behavior with copied page titles, which is a change from how things used to work in Wagtail 3.

Reply:
Thanks for the feedback @laymonage! Have tried your suggested code, and the behaviour of `draft_title` does now seem to match the Wagtail 3.x behaviour - however, I notice that when the copy is performed with `keep_live=True`, draft_title fails to update, on both Wagtail 3.x and the new code. Presumably that's the unwanted side-effect of `page_copy.save` you mention above. Will need to do some more digging to figure out whether that's something we should (and can) fix...

Reply:
Ah, yep, forgot to mention that – your fix is just as what I had in mind, though I didn't try it myself. Have now just tested with your latest commit and it seems to work great.